### ISSUE TITLE
SimplifyBooleanExpressionCheck throw NullPointerException while checking ternary operator

### ISSUE THREAD

**Issue by SebHeuze** [checkstyle/checkstyle#13140]
Since #11020 we have a problem with a project that contain ternary operator with unnecessary parenthesis
This is because the code introduced in this PR lack of a null checking and throw :

```
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing 
...
Caused by: java.lang.NullPointerException: Cannot invoke "com.puppycrawl.tools.checkstyle.api.DetailAST.getType()" because the return value of "com.puppycrawl.tools.checkstyle.api.DetailAST.getNextSibling()" is null
```

Here is an example to reproduce the problem

```bash
/var/tmp $ javac CheckStyleBug.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
  <property name="severity" value="error"/>
  <property name="fileExtensions" value="java, properties, xml"/>
  <module name="TreeWalker">
    <module name="SimplifyBooleanExpression"/>
  </module>
</module>

/var/tmp $ cat CheckStyleBug.java
package checkstyleTest;

public class CheckStyleBug {

        public static void main(String[] args) {
       Boolean value = null;
       boolean temp = value != null ? value : (false); // ok
        }

}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java %RUN_LOCALE% -jar checkstyle-10.12.0-all.jar -c config.xml CheckStyleBug.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing CheckStyleBug.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:306)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:223)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:415)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.NullPointerException
        at com.puppycrawl.tools.checkstyle.checks.coding.SimplifyBooleanExpressionCheck.visitToken(SimplifyBooleanExpressionCheck.java:126)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:335)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:406)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:273)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:154)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:334)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:293)
        ... 5 more
```

**Describe what you expect in detail.**
---
Analysis should not throw NullPointerException

**Comment by nrmancuso** [comment]
Please follow issue template

**Comment by SebHeuze** [comment]
Edited the issue

**Comment by nrmancuso** [comment]
Closed via #13141
